src/diffusers/loaders.py

```diff
@@ -1584,19 +1596,30 @@ def load_lora_into_text_encoder(
                 if cls.use_peft_backend:
                     from peft import LoraConfig
 
-                    lora_rank = list(rank.values())[0]
-                    # By definition, the scale should be alpha divided by rank.
-                    # https://github.com/huggingface/peft/blob/ba0477f2985b1ba311b83459d29895c809404e99/src/peft/tuners/lora/layer.py#L71
-                    alpha = lora_scale * lora_rank
+                    r, lora_alpha, rank_pattern, alpha_pattern, target_modules = get_rank_and_alpha_pattern(
+                        rank, network_alphas, text_encoder_lora_state_dict
+                    )
```

Comment:
What's `r`? Could we use a better variable name here?

Reply:
Also, what's meant by `pattern`? To help the contributors, I would not mind adding a comment to explain what "pattern" here means.

Reply:
> What's r? Could we use a better variable name here?

Rank. the naming was done to match the parameters accepted by the `LoraConfig`.

> Also, what's meant by pattern? To help the contributors, I would not mind adding a comment to explain what "pattern" here means.

Makes sense


https://github.com/huggingface/diffusers/pull/5147#discussion_r1335993659

If the above is done, all this will be replaced. So, will be doing the above.

Reply:
addressed in the latest commit